lib/web_ui/lib/src/engine/navigation/history.dart

```diff
@@ -62,7 +62,11 @@ abstract class BrowserHistory {
   /// The state of the current location of the user's browser.
   Object? get currentState => urlStrategy?.getState();
 
-  /// Update the url with the given [routeName] and [state].
+  /// Update the url with the given `routeName` and `state`.
+  /// 
+  /// If the `replace` is false, the caller wants to push a new route name
```

Comment:
not sure what language we use elsewhere, but should it be "history entry" instead of "route name" since its more than just the route name and its not necessarily a new route name (could be the same with different state?).

Reply:
history entry may not make sense for some of the subclass. For example the singleEntry one doesn't expose the idea of history entry. I will change it to routeName and state